sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/batch/BatchExecutor.java

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos.implementation.batch;
+
+import com.azure.cosmos.BridgeInternal;
+import com.azure.cosmos.CosmosAsyncContainer;
+import com.azure.cosmos.CosmosBridgeInternal;
+import com.azure.cosmos.TransactionalBatch;
+import com.azure.cosmos.TransactionalBatchRequestOptions;
+import com.azure.cosmos.TransactionalBatchResponse;
+import reactor.core.publisher.Mono;
+
+import java.util.List;
+
+import static com.azure.cosmos.implementation.guava25.base.Preconditions.checkArgument;
+
+public final class BatchExecutor {
+
+    private final CosmosAsyncContainer container;
+    private final TransactionalBatchRequestOptions options;
+    private final TransactionalBatch transactionalBatch;
+
+    public BatchExecutor(
+        final CosmosAsyncContainer container,
+        final TransactionalBatch transactionalBatch,
+        final TransactionalBatchRequestOptions options) {
+
+        this.container = container;
+        this.transactionalBatch = transactionalBatch;
+        this.options = options;
+    }
+
+    /**
+     * Create a batch request from list of operations and executes it.
+     *
+     * @return Response from the server.
+     */
+    public final Mono<TransactionalBatchResponse> executeAsync() {
```

Comment:
Please make sure that the user can call `executeAsync` multiple time if 429 or other transient failure is returned.

Reply:
It is inside the context and every time block is called, executeAsync will be called.

Reply:
Yeah - the github issue for .Net is a non-issue here.